src/cedict-service/cedict.js

```diff
@@ -0,0 +1,389 @@
+  }
+
+  /**
+   * Initializes a data object.
+   *
+   * @returns {Promise<undefined> | Promise<Error>} - A promise
+   */
+  init () {
+    return new Promise((resolve, reject) => {
+      this._storage = new CedictPermanentStorage(this._configuration.storage)
+      // `storage.connect()` will create a database if it does not exist yet.
+      return this._storage.connect()
+        .catch((error) => {
+          console.error('Connection to storage cannot be established')
+          reject(error)
+        })
+        .then(() => this._storage.getIntergrityData())
+        .then((storageData) => {
+          /*
+          Integrity data has been returned successfully which means database structure is OK.
+          Let's check if there is a new version of data available on a server.
+           */
+          if (
+            storageData.recordsInMeta !== 1 ||
+            storageData.recordsInDictionary !== this._configuration.data.recordsCount ||
+            storageData.metadata.version !== this._configuration.data.version ||
+            storageData.metadata.revision !== this._configuration.data.revision
+          ) {
+            throw new Error('Storage is outdated')
+          }
+          // Data in storage is fresh so we can read it into memory structures if we have that option enabled
+          if (this._configuration.storage.stores.dictionary.volatileStorage.enabled) {
+            return this._storage.stores.dictionary.getAllEntries()
+              .then((entries) => this.populateVolatileStorage(storageData.metadata, entries))
+              .catch((error) => reject(error))
+          }
+        })
+        .catch(() => {
+          // Data in permanent storage needs to be updated
+          return this._storage.destroy()
+            // `connect()` will create storage and stores
+            .then(() => this._storage.connect())
+            .then(() => this.downloadData())
+            .then(({ meta, dictionary }) => {
+              if (this._configuration.storage.stores.dictionary.volatileStorage.enabled) {
+                this.populateVolatileStorage(meta, dictionary)
+              }
+              return this.populatePermanentStorage(meta, dictionary)
+                .then(() => Promise.resolve())
+                .catch((error) => {
+                  console.error('Unable to store CEDICT data to IndexedDB.', error)
+                  // Cannot write CEDICT data to IndexedDB. Will fall back to an in-memory location
+                  if (!this._configuration.storage.stores.dictionary.volatileStorage.enabled) {
+                    console.warn('Switched to in-memory placement of CEDICT data')
+                    this._configuration.storage.stores.dictionary.volatileStorage.enabled = true
+                    this.populateVolatileStorage(meta, dictionary)
+                  }
+                  return Promise.resolve()
+                })
+            })
+        })
```

Comment:
It will catch an exception thrown on line 123 and will recreate database and download CEDICT data